src/com/xilinx/rapidwright/util/StringTools.java

```diff
@@ -159,6 +159,20 @@ public static String removeLastSeparator(String s) {
         return s;
     }
 
+    /**
+     * Replaces the file extension of the provided file name. The current extension
+     * includes the final period '.' and all characters following it. If the file
+     * has no extension, it will add it as a suffix to the filename.
+     * 
+     * @param fileName     Name of the file to receive the updated extension.
+     * @param newExtension The new extension (should include starting period '.')
+     * @return The newly updated file name.
+     */
+    public static String replaceExtension(String fileName, String newExtension) {
```

Comment:
Are you aware we have a `FileTools.replaceExtension(Path, String)` method (which admittedly has a different signature to this). Would it make sense to collect them together?

Reply:
I thought there was one but couldn't find it.  I've abandoned the `StringTools` version.